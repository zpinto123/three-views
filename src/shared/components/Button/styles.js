import { css } from "styled-components";
const styles = {
  button: css`
    padding-left: 10px !important;
    padding-right: 10px !important;
  `
};

export default styles;
