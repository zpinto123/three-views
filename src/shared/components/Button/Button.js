import React from "react";
import PropTypes from "prop-types";
import { Button as MUIButton } from "@material-ui/core";
import { styleUtils } from "../../utils";
import styles from "./styles";

const { style } = styleUtils;
const SButton = style(MUIButton, styles.button);

const Button = ({ handleClick, children, size, color, ...restOfProps }) => (
  <SButton onClick={handleClick} color={color} size={size} {...restOfProps}>
    {children}
  </SButton>
);

Button.propTypes = {
  size: PropTypes.string,
  color: PropTypes.string,
  children: PropTypes.string.isRequired,
  handleClick: PropTypes.func.isRequired
};

Button.defaultProps = {
  size: "small",
  color: "primary"
};

export default Button;
