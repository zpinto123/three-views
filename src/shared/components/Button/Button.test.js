import React from "react";
import renderer from "react-test-renderer";
import { mount } from "enzyme";
import "jest-styled-components";

import Button from "./";

describe("<Button />", () => {
  it("Show Button correctly", () => {
    const mockCallBack = jest.fn();

    const component = renderer.create(
      <Button handleClick={mockCallBack}>Button test</Button>
    );
    expect(component.toJSON()).toMatchSnapshot();
  });

  it("onClick works correcly", () => {
    const mockCallBack = jest.fn();

    const component = mount(<Button handleClick={mockCallBack}>Ok!</Button>);
    component.find("button").simulate("click");
    expect(mockCallBack.mock.calls.length).toEqual(1);
  });
});
