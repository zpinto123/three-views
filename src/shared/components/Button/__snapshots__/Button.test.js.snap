// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> Show Button correctly 1`] = `
.c0 {
  padding-left: 10px !important;
  padding-right: 10px !important;
}

<button
  className="MuiButtonBase-root MuiButton-root c0 MuiButton-text MuiButton-textPrimary MuiButton-sizeSmall"
  disabled={false}
  onBlur={[Function]}
  onClick={[MockFunction]}
  onDragLeave={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  tabIndex={0}
  type="button"
>
  <span
    className="MuiButton-label"
  >
    Button test
  </span>
</button>
`;
